src/Sound.c

```diff
@@ -37,30 +56,53 @@ Mix_Chunk* _sample[8];
  */
 bool TLN_SoundInit(int nsounds, char **filenames)
 {
+	char path[MAX_PATH + 1];
 	memset(_sample, 0, sizeof(Mix_Chunk*) * 8);
+	if (!TLN_AudioSybsysInit())
+		return false;
 
-    // Set up the audio stream
-    if(Mix_OpenAudio(MIX_DEFAULT_FREQUENCY, MIX_DEFAULT_FORMAT, MIX_DEFAULT_CHANNELS, 1024) < 0)
-    {
+	// Load waveforms
+	for(int i = 0; i < nsounds; i++)
+	{
+#if (_MSC_VER) && (_MSC_VER < 1900)
+		sprintf (path, "%s/%s", TLN_GetLoadPath(), filenames[i]);
+#else
+		snprintf (path, sizeof(path), "%s/%s", TLN_GetLoadPath(), filenames[i]);
+#endif
+		_sample[i] = Mix_LoadWAV(path);
```

Comment:
This does not work because the size of the array `_sample` is 8 not of `nsounds`